xrdp/xrdp_main_utils.c

```diff
@@ -0,0 +1,252 @@
+        /* set this event so that the main thread know if
+         * g_process_waiting_function() must be called */
+        g_set_wait_obj(g_sync_event);
+
+        do
+        {
+            g_sleep(100);
+            tc_mutex_lock(g_sync_mutex);
+            /* load new value from global to see if the g_process_waiting_function()
+             * function has processed the function */
+            sync_command = g_sync_command;
+            sync_result = g_sync_result;
+            tc_mutex_unlock(g_sync_mutex);
+        }
+        while (sync_command != 0); /* loop until g_process_waiting_function()
+                                * has processed the request */
+        tc_mutex_unlock(g_sync1_mutex);
+        LOG_DEVEL(LOG_LEVEL_DEBUG, "g_xrdp_sync processed BY main thread -> continue");
+    }
+
+    return sync_result;
+}
+
+/*****************************************************************************/
+/* called in child just after fork */
+int
+xrdp_child_fork(void)
+{
+    int pid;
+    char text[256];
+
+    /* close, don't delete these */
+    g_close_wait_obj(g_term_event);
+    g_close_wait_obj(g_sync_event);
+    pid = g_getpid();
+    g_snprintf(text, 255, "xrdp_%8.8x_main_term", pid);
+    g_term_event = g_create_wait_obj(text);
+    g_snprintf(text, 255, "xrdp_%8.8x_main_sync", pid);
+    g_sync_event = g_create_wait_obj(text);
+    return 0;
+}
+
+/*****************************************************************************/
+long
+g_get_g_sync_mutex(void)
+{
+    return g_sync_mutex;
+}
+
+/*****************************************************************************/
+void
+g_set_g_sync_mutex(long mutex)
+{
+    g_sync_mutex = mutex;
+}
+
+/*****************************************************************************/
+long
+g_get_g_sync1_mutex(void)
+{
+    return g_sync1_mutex;
+}
+
+/*****************************************************************************/
+void
+g_set_g_sync1_mutex(long mutex)
+{
+    g_sync1_mutex = mutex;
+}
+
+/*****************************************************************************/
+tbus
+g_get_g_term_event(void)
+{
+    return g_term_event;
+}
+
+/*****************************************************************************/
+void
+g_set_g_term_event(tbus event)
+{
+    g_term_event = event;
+}
+
+/*****************************************************************************/
+tbus
+g_get_g_sync_event(void)
+{
+    return g_sync_event;
+}
+
+/*****************************************************************************/
+void
+g_set_g_sync_event(tbus event)
+{
+    g_sync_event = event;
+}
+
+/*****************************************************************************/
+long
+g_get_threadid(void)
+{
+    return g_threadid;
+}
+
+/*****************************************************************************/
+void
+g_set_threadid(long id)
+{
+    g_threadid = id;
+}
+
+/*****************************************************************************/
+int
+g_get_term(void)
```

Comment:
Don't need this and `g_get_g_term_event()` - they look identical.

Reply:
Turns out I also don't need `g_get_term_event` either!

Removed and consolidated.